src/SymbolTable.ts

```diff
@@ -457,3 +703,22 @@ export interface TypeCacheEntry {
     data?: ExtraSymbolData;
     flags?: SymbolTypeFlag;
 }
+
+export interface PocketTable {
+    table: SymbolTable;
+    /**
+     * The index of the statement that contains this table within its parent block
+     */
+    index: number;
+    /**
+     * Is it mandatory for code to flow through this pocket?
+     */
+    isMandatory?: boolean;
```

Comment:
I also don't love `isMandatory` ... I want a flag that shows that a particular pocket table *Must* be used.